posts/2020-09-25-signed-certificate-with-acme-200010.adoc

```diff
@@ -0,0 +1,135 @@
+
+To use the ACME feature, provide the directory URI for the ACME Certificate Authority (CA) and the domain name for the server. Public ACME providers call back on port 80 to verify domain ownership before issuing a certificate. When the Open Liberty server starts, the provided CA directory URI will be used to request a certificate. The CA signed certificate is installed in the keystore and acts as the default certificate.
+
+Add the feature to the `server.xml`:
+[source, xml]
+----
+<featureManager> 
+    <feature>acmeCA-2.0</feature>
+</featureManager>
+
+<acmeCA directoryURI="https://acme.host.com/directory" >
+    <domain>theDomainThatIOwn.com</domain>
+   <accountContact>mailto:my_email_addr@theDomainThatIOwn.com</accountContact>
+</acmeCA>
+
+<httpEndpoint host="*" httpPort="80" httpsPort="443" id="defaultHttpEndpoint"/>
+<keyStore password="password_for_keystore" id="defaultKeyStore"/>
+----
+
+For more information:
+
+* https://en.wikipedia.org/wiki/[High level summary of the ACME protocol]
+* https://tools.ietf.org/html/rfc8555[The ACME spec]
+* https://letsencrypt.org/[Popular public ACME CA]
+
+
+[#bugs]
+== Significant bugs fixed in this release
+
+We’ve spent some time fixing bugs. The following sections describe just some of the issues we resolved in this release. If you’re interested, here's the full list of link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A200010+label%3A%22release+bug%22+[fixed bugs in 20.0.0.10].
+
+=== Enabling openTracing with no tracer class configured
+
+When the MicroProfile 3.x feature is enabled, `mpOpenTracing-1.3` is also enabled even if you have no intention of using OpenTracing. Most likely the tracer fails to load because there is a classloader leak in the OpenTracerManager. Previously, mpOpenTracing-1.3 code tried to load the tracer on every JAX-RS request. This issue has been fixed by loading the tracer only once. A dummy tracer is stored in the tracer map when the initial loading of the tracer fails. Subsequent requests do not load the tracer again when the map has a dummy tracer. Also `AppStateListener` is added to the listener when the application stops. The corresponding entry in the tracer map is removed. The `AppStateListener` also listens to `<Library />`` changes. All entries in the tracer map is cleared.
```

Comment:
When the MicroProfile 3.x feature is enabled, `mpOpenTracing-1.3` is also enabled even if you have no intention of using OpenTracing. Most likely the tracer fails to load because it is not added to the classpath. Previously, `mpOpenTracing-1.3` code tried to load the tracer on every JAX-RS request. This logic has been improved by loading the tracer only once to improve efficiency. Subsequent requests do not load the tracer again. Also the tracer loaded for the application will be removed when the application stops.

Reply:
Thank you, that's been changed